### ISSUE TITLE
ARIA/CTR mode self test failures

### ISSUE THREAD

**Issue by noloader** [weidai11/cryptopp#1235]
Rabbit, RabbitWithIV, HC128 and HC256 were producing incorrect results when `inString == outString` as reported in [GH #1231](https://github.com/weidai11/cryptopp/issues/1231). We added self tests to detect the failures, and cleared the failures.

The new tests also revealed ARIA/CTR mode also has problems.

This bug will track ARIA/CTR.

**Comment by noloader** [comment]
This was cleared in two steps. The 2nd step, which fixed the issue, is [Commit dde8e9fa23bc](https://github.com/weidai11/cryptopp/commit/dde8e9fa23bc). The dde8e9fa23bc commit removed this block, and folded it into building `outBlock`.

```diff
-	if (xorBlock != NULLPTR)
-		for (unsigned int n=0; n<ARIA::BLOCKSIZE; ++n)
-			outBlock[n] ^= xorBlock[n];
```

The 1st commit, [Commit 5250ab2bf2b0](https://github.com/weidai11/cryptopp/commit/5250ab2bf2b0), removed the SIMD code since it was no longer needed due to the second commit.

**Comment by noloader** [comment]
> I do not know why this fixed ARIA. The two codes should have been equivalent.

ARIA had undefined behavior. Ugh!

The working area - `m_w` - was too small. It should have been a `SecWordBlock` with room for 28 word-sized elements. Instead it only had room for 17 elements. Arg!!!

Even more baffling, Valgrind never alerted to the problem.

Also see [Commit d3d23002f607](https://github.com/weidai11/cryptopp/commit/d3d23002f607).
